core/arch/arm/kernel/boot.c

```diff
@@ -1266,6 +1273,29 @@ void __weak boot_init_primary_early(void)
 		mapped_tl = transfer_list_map(boot_arg_transfer_list);
 		if (!mapped_tl)
 			panic("Failed to map transfer list");
+
+		transfer_list_dump(mapped_tl);
+		tl_e = transfer_list_find(mapped_tl, TL_TAG_FDT);
+		if (IS_ENABLED(CFG_EXTERNAL_DT) && tl_e) {
+			/*
+			 * Expand the data size of the DTB entry to the maximum
+			 * allocable mapped memory to reserve sufficient space
+			 * for inserting new nodes.
+			 */
+			uint32_t dtb_max_sz = mapped_tl->max_size -
+					      mapped_tl->size + tl_e->data_size;
+
+			if (dtb_max_sz > CFG_DTB_MAX_SIZE)
+				dtb_max_sz = CFG_DTB_MAX_SIZE;
+
+			if (!transfer_list_set_data_size(mapped_tl, tl_e,
+							 dtb_max_sz)) {
+				EMSG("Failed to extend DTB size to %#"PRIx32,
+				     dtb_max_sz);
+				panic();
+			}
+			mapped_dt_e = tl_e;
```

Comment:
It seems that the purpose of `mapped_dt_e` is to cache this entry to avoid calling `transfer_list_find(mapped_tl, TL_TAG_FDT)`. I would rather avoid yet another global boot variable.

Reply:
All right, I can remove this variable.